xml_converter/generators/code_generator.py

```diff
@@ -286,8 +295,10 @@ class AttributeVariable:
     xml_fields: List[str]
     protobuf_field: str
     default_xml_fields: List[str] = field(default_factory=list)
-    xml_export: str = ""
-    is_child: bool = False
+    xml_child_export: List[str] = field(default_factory=list)
+    xml_parent_export: str = ""
+    compound_name: Optional[str] = None
+    is_xml_export: bool = True
```

Comment:
I had the thought, since this variable is being used to determine if the attribute should be written to the xml file, the name should be `write_to_xml` with a default True. I also found that I can change the code so that components now have the attribute name `parent.component` and added a separate variable for `set_flag_name` which is usually just the parent name. But by doing this it greatly simplified the template file for `init_xml_attribute()` and `as_xml()`.